crates/circuit/src/lib.rs

```diff
@@ -35,9 +35,40 @@ use pyo3::types::{PySequence, PyTuple};
 pub type BitType = u32;
 #[derive(Copy, Clone, Debug, Hash, Ord, PartialOrd, Eq, PartialEq, FromPyObject)]
 pub struct Qubit(pub BitType);
+
+impl Qubit {
+    /// Construct a new Qubit object from a usize, if you have a u32 you can
+    /// create a `Qubit` object directly with `Qubit(0u32)`.
+    #[inline(always)]
+    pub fn new(index: usize) -> Self {
+        Qubit(index as BitType)
```

Comment:
We might want to do something like this instead, since otherwise if an `index` is given that is too big for `BitType`, [Rust will wrap it](https://doc.rust-lang.org/rust-by-example/types/cast.html#casting).


```suggestion
        Qubit(index.try_into().expect(format!("Index value '{}' exceeds the maximum bit width!", index))
```

Reply:
I based the behavior here on that petgraph does for it's `NodeIndex` and `EdgeIndex` types which are essentially the same (except their generic instead of having a fixed inner type) and they do the ceiling with `as`. But I agree panicking in case of an overflow makes more sense. I'll apply the suggestion and add a test checking for the panic.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/13284/commits/1e6d4af8cdc8518e9b0a2ed3a7a725280ea90ab8